backend/src/main/kotlin/hu/bme/sch/cmsch/component/login/LoginService.kt

```diff
@@ -129,22 +133,47 @@ open class LoginService(
         }
 
         // Update user profile values
-        if (profile.neptun != null)
+        if (profile.neptun != null){
             user.neptun = profile.neptun ?: user.neptun
-        if (profile.email != null && profile.email?.isNotBlank() == true)
+            user.detailsImported = true
+        }
+        if (profile.email != null && profile.email?.isNotBlank() == true){
             user.email = profile.email ?: user.email
+            user.detailsImported = true
+        }
+
 
         // Check neptun; grant group and guild if mapping present
         if (user.neptun.isNotBlank() && user.groupName.isBlank()) {
             groupToUserMapping.findByNeptun(user.neptun).ifPresent {
                 user.major = it.major
                 user.groupName = it.groupName
                 user.group = groups.findByName(it.groupName).orElse(null)
+                user.detailsImported = true
+            }
+        }
+
+        if(user.internalId.isNotBlank() && !user.detailsImported){
```

Comment:
Code smell: We use spaces between the ifs and the condition bracket.

Reply:
Set the `user.detailsImporte` at the end to avoid skipping this segment all the times. Eg. `(profile.email != null && profile.email?.isNotBlank() == true` will always be true in practice.

Reply:
Will email and neptun  always imported in practice? If only these then removing the `user.detailsImported` setting from line 138 and 142 fixes this problem, am I right?

Reply:
We can optionally import email and neptun code via authsch. We want to import some other details (using internalId for example; this feature is all about this). I'd only set `user.detailsImported` to `true` if it is imported by your new feature.